### PR TITLE
[#417] refactor: Eliminate raw use of parameterized class

### PR DESCRIPTION
**Pull request by cchung100m**
Hi @zuston @advancedxy

Any comments and suggestions would be appreciated if you are available, thank you.

### What changes were proposed in this pull request?

Eliminate `raw use of parameterized class`

### Why are the changes needed?

Fix: #417

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

current UT

### CODE REVIEW

File: coordinator/src/main/java/org/apache/uniffle/coordinator/web/servlet/BaseServlet.java [hunk 1]
```diff
@@ -44,9 +44,9 @@ protected void doPost(HttpServletRequest req, HttpServletResponse resp) throws I
     writeJSON(resp, handlerRequest(() -> handlePost(req, resp)));
   }
 
-  private Response handlerRequest(
-      Callable<Response> function) {
-    Response response;
+  private <T> Response<T> handlerRequest(
```
[the remark this answers is not in the document]
Comment (xianjingfeng) [reply coordinator/src/main/java/org/apache/uniffle/coordinator/web/servlet/BaseServlet.java:47]: How about Response<Object>？

[the remark this answers is not in the document]
Comment (xianjingfeng) [reply coordinator/src/main/java/org/apache/uniffle/coordinator/web/servlet/BaseServlet.java:47]: How about Response&lt;Object>?

File: common/src/main/java/org/apache/uniffle/common/util/RetryUtils.java [hunk 2]
```diff
@@ -32,7 +32,7 @@ public static <T> T retry(RetryCmd<T> cmd, long intervalMs, int retryTimes) thro
   }
 
   public static <T> T retry(RetryCmd<T> cmd, long intervalMs, int retryTimes,
-      Set<Class> exceptionClasses) throws Throwable {
+      Set<Class<? extends Object>> exceptionClasses) throws Throwable {
```
[the remark this answers is not in the document]
Comment (cchung100m) [reply common/src/main/java/org/apache/uniffle/common/util/RetryUtils.java]: Hi @advancedxy 
Thanks for the kind reminder, I updated the part you mentioned.

File: coordinator/src/main/java/org/apache/uniffle/coordinator/web/servlet/BaseServlet.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (xianjingfeng) [reply coordinator/src/main/java/org/apache/uniffle/coordinator/web/servlet/BaseServlet.java:47]: No rules,We can implement all REST api through one servlet.

[the remark this answers is not in the document]
Comment (cchung100m) [reply coordinator/src/main/java/org/apache/uniffle/coordinator/web/servlet/BaseServlet.java:47]: Hi @advancedxy  @xianjingfeng 

Thank you for sharing your thoughts and active feedback.  I am open to whether make `BaseServlet` a generic class or not.  Willing to do the change if we have a conclusion.

**Review by jerqi — APPROVED** [review]
LGTM, thanks all

**Pull request merged**